build.py

```diff
@@ -1079,6 +1099,38 @@ def dockerfile_prepare_container_linux(argmap, backends, enable_gpu,
 ENV TCMALLOC_RELEASE_RATE 200
 '''.format(gpu_enabled=gpu_enabled, backend_dependencies=backend_dependencies)
 
+    if ('fastertransformer' in backends):
+        be = 'fastertransformer'
+        github_link = FLAGS.github_organization + '/' + be + '_backend.git'
+        dir_name = "/tmp/tmp_" + be
+        # clone the fastertransformer repo and copy over create_dockerfile.py
+        op = ['mkdir', '-p', dir_name]
+        msg = 'Creating repo {} failed.'.format(dir_name)
+        run_subprocess(op, msg)
+        op = ['rm', '-rf', dir_name]
+        msg = 'Removing repo {} failed.'.format(dir_name)
+        run_subprocess(op, msg)
+        op = [
+            'git', 'clone', '--single-branch', '--depth=1', '-b', backends[be],
+            github_link, dir_name
+        ]
+        msg = 'git clone of branch {}:{} failed'.format(github_link,
+                                                        backends[be])
+        run_subprocess(op, msg)
+
+        op = ['cp', dir_name + "/docker/create_dockerfile_and_build.py", '.']
+        msg = 'Copy dockerfile to current repo failed.'
+        run_subprocess(op, msg)
+
+        import create_dockerfile_and_build
+        df += create_dockerfile_and_build.create_postbuild(
+            is_multistage_build=False)
+
+        # clean up
+        op = ['rm', '-r', 'create_dockerfile_and_build.py']
+        msg = 'Failed to remove directory {}'.format(dir_name)
+        run_subprocess(op, msg)
```

Comment:
So the fastertransformer backend is built at the last docker stage? Based on the comment it should not contain the backends:
```
Production stage: Create container with just inference server executable
```
Why can't fastertransformer follow our guideline to encapsulate the build in CMake so it can utilize the same `backend_build` step in `build.py`?

Reply:
@GuanLuo we do build it with the same `backend_build` with `def fastertransformer_cmake_args():`. I'm simply running the few commands required in the post-process phase:
```
     df = '''
ENV WORKSPACE /workspace
WORKDIR /workspace       
'''
    df += '''
ENV NCCL_LAUNCH_MODE=GROUP
RUN sed -i 's/#X11UseLocalhost yes/X11UseLocalhost no/g' /etc/ssh/sshd_config \\
    && mkdir /var/run/sshd -p
    '''
```

Reply:
https://github.com/triton-inference-server/fastertransformer_backend/blob/main/docker/create_dockerfile_and_build.py

Reply:
I was wondering if we can prevent this additional clone process. How about copying out create_dockerfile_and_build.py file from the build step of FasterTransformer backend?

Reply:
Thanks for the clarification. Given the intention, what if you directly fetch the file content (HTTP GET versioned/path/to/create_dockerfile_and_build.py) and [construct Python module in place](https://stackoverflow.com/questions/5362771/how-to-load-a-module-from-code-in-a-string)? Then there is no exchange outside the script process